src/server/transaction.cc

```diff
@@ -941,22 +869,33 @@ void Transaction::ExecuteAsync() {
   // initialize the run barrier before arming, as well as copy indices
   // of active shards to avoid reading concurrently accessed shard data.
   std::bitset<1024> poll_flags(0);
-  run_barrier_.Start(unique_shard_cnt_);
+  IterateActiveShards([&poll_flags, global = IsGlobal() || IsAtomicMulti()](auto& sd, auto i) {
+    if ((sd.local_mask & RAN_IMMEDIATELY) == 0)
+      poll_flags.set(i, true);
+    sd.local_mask &= ~RAN_IMMEDIATELY;  // we'll run it next time if it avoided concluding
+  });
+
+  unsigned run_cnt = poll_flags.count();
```

Comment:
for a single thread , the loop above will run once, but this operation will still have to touch all the pages in bitset. maybe just increment the counter in the loop?

Reply:
Ok, will change. That's why I didn't like a big bitset either 😆 But it likely has no effect, its a loop over 16 elements